src/lib.rs

```diff
@@ -216,8 +216,7 @@ impl<T: Send> ThreadLocal<T> {
         }
         unsafe {
             let entry = &*bucket_ptr.add(thread.index);
-            // Read without atomic operations as only this thread can set the value.
-            if (&entry.present as *const _ as *const bool).read() {
+            if entry.present.load(Ordering::Acquire) {
```

Comment:
This should be `Relaxed` since there are no ordering constraints with other threads.

Reply:
The iterator uses Acquire, is it OK not to use that here?

Reply:
The iterator will access elements of other threads, but here we are only accessing the element for our thread.

Reply:
Done!